ui/client/reducers/graph.js

```diff
@@ -414,3 +421,54 @@ function createEdge(fromNode, toNode, edgeType, allEdges, processDefinitionData)
   const adjustedEdgeType = edgeType || NodeUtils.edgeType(allEdges, fromNode, processDefinitionData)
   return adjustedEdgeType ? {...baseEdge, edgeType: adjustedEdgeType} : baseEdge
 }
+
+function adjustBranchParametersAfterDisconnect(nodes, removedEdgeFrom, removedEdgeTo) {
+  const node = nodes.find(n => n.id === removedEdgeTo)
+  if (node && NodeUtils.nodeIsJoin(node)) {
+    const newToNode = removeBranchParameter(node, removedEdgeFrom)
+    return _.map(nodes, (n) => { return n.id === removedEdgeTo ? newToNode : n })
+  } else {
+    return nodes
+  }
+
+}
+
+function removeBranchParameter(node, branchId) {
+  const clone = _.cloneDeep(node)
+  clone.branchParameters = _.reject(clone.branchParameters, parameter => parameter.branchId === branchId)
+  return clone
+}
+
+function enrichNodeWithProcessDependentData(originalNode, processDefinitionData, edges) {
+  const node = _.cloneDeepWith(originalNode)
+  if (NodeUtils.nodeIsJoin(node)) {
+    const nodeObjectDetails = ProcessUtils.findNodeObjectTypeDefinition(node, processDefinitionData.processDefinition)
+    const declaredBranchParameters = nodeObjectDetails.parameters.filter(p => p.branchParam)
+    const incomingEdges = edges.filter(e => e.to === node.id)
+
+    node.branchParameters = incomingEdges.map((edge) => {
+      let branchId = edge.from
+      let existingBranchParams = node.branchParameters.find(p => p.branchId === branchId)
+      let newBranchParams = declaredBranchParameters.map((branchParamDef) => {
+        let existingParamValue = ((existingBranchParams || {}).parameters || []).find(p => p.name === branchParamDef.name)
+        let templateParamValue = (node.branchParametersTemplate || []).find(p => p.name === branchParamDef.name)
+        return existingParamValue || _.cloneDeep(templateParamValue) ||
+            // We need to have this fallback to some template for situation when it is existing node and it has't got
+            // defined parameters filled. see note in DefinitionPreparer on backend side TODO: remove it after API refactor
+            _.cloneDeep({
+              name: branchParamDef.name,
+              expression: {
+                expression: `#${branchParamDef.name}`,
+                language: "spel",
+              },
+            })
+      })
+      return {
+        branchId: branchId,
+        parameters: newBranchParams,
+      }
+    })
+    delete node["branchParametersTemplate"]
```

Comment:
From my tests it looks like now we shouldn't delete template, because this node will be reused during another connection adding...

Reply:
good catch, done